File: index/index_create.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

engines = [
        {
          "name": "marengo2.6",
          "options": ["visual", "conversation", "text_in_video", "logo"]
        },
        {
            "name": "pegasus1.1",
            "options": ["visual", "conversation"]
        }
    ]
created_index = client.index.create(
    name="",
    engines=engines,
    addons=["thumbnail"]
)
print(f"ID: {created_index.id}")
print(f"Name: {created_index.name}")
print("Engines:")
for i, engine in enumerate(created_index.engines, 1):
    print(f"  Engine {i}:")
    print(f"    Name: {engine.name}")
    print(f"    Options: {engine.options}")
print(f"Video count: {created_index.video_count}")
print(f"Total duration: {created_index.total_duration} seconds")
print(f"created At: {created_index.created_at}")
if created_index.updated_at:
    print(f"Updated at: {created_index.updated_at}")

File: index/index_create.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 

  const createdIndex = await client.index.create({
    name: '',
    engines: [
      {
        name: 'marengo2.6',
        options: ['visual', 'conversation', 'text_in_video']
      },
      {
        name: 'pegasus1.1',
        options: ['visual', 'conversation']
      }
    ],
    addons: ['thumbnail'],
  });
  console.log(`ID: ${createdIndex.id}`);
  console.log(`Name: ${createdIndex.name}`);
  console.log("Engines:");
  createdIndex.engines.forEach((engine, index) => {
    console.log(`  Engine ${index + 1}:`);
    console.log(`    Name: ${engine.name}`);
    console.log(`    Options: ${JSON.stringify(engine.options)}`);
  });
  console.log(`Video count: ${createdIndex.videoCount}`);
  console.log(`Total turation: ${createdIndex.totalDuration} seconds`);
  console.log(`Created at: ${createdIndex.createdAt}`);
  if (createdIndex.updatedAt) {
    console.log(`Updated at: ${createdIndex.updatedAt}`);
  }
})();
